### ISSUE TITLE
 Feature request: support of rooms

### ISSUE THREAD

**Issue by minchopm** [gondar00/yjs-websocket-server-with-nestjs#1]
Is there any way to implement support of rooms

**Comment by gondar00** [comment]
The second parameter [here](https://github.com/gondar00/yjs-websocket-server-with-nestjs/blob/master/client/src/y.ts#L8) while creating the y-websocket connection is the [room](https://github.com/yjs/y-websocket/blob/master/src/y-websocket.js#L242) name

**Comment by jblyberg** [comment]
When I pass a room name, I'm getting, `Firefox can’t establish a connection to the server at ws://localhost:5001/testroom.`

It seems that the connection request will not even pass `handleConnection()` in the nest gateway.

**Comment by gondar00** [comment]
@jblyberg you're right, I found another way where you can set the `roomName` in the [cookie](https://github.com/gondar00/yjs-websocket-server-with-nestjs/blob/master/client/src/app.tsx#L11) and connect users to different [yjs](https://github.com/gondar00/yjs-websocket-server-with-nestjs/blob/master/src/yjs.gateway.ts#L36) documents based on that value

**Comment by minchopm** [comment]
You could pass room name as param

```
new WebsocketProvider(this.socketPath, '', yDoc, {
     params: {'roomName': subDoc.guid}
});
```

and read it 

```
const parsedUrl = url.parse(req.url);
const parsedQs = querystring.parse(parsedUrl.query);
if(parsedQs.roomName){
    const docName = parsedQs.roomName;
    setupWSConnection(connection, req, { ...(docName && { docName }) });
}

```
if you handle multiple Web socket connections on same page, will be difficult to manage their rooms

**Comment by titanbit** [comment]
Is there any way to connect with nestjs application with dynamic roomIds, my client side is running on react and backend is running on nestjs. but unable to connect to the room id. 

frontend code : 
      provider.current = new WebsocketProvider(
        `ws://localhost:3030/ws/`, // Replace with your WebSocket server URL
        `${fileId}`,
        ydoc.current,
      )
      
   Backend : 
   
   import {
  WebSocketGateway,
  OnGatewayConnection,
  OnGatewayDisconnect,
  WebSocketServer,
} from '@nestjs/websockets';
// import { WebSocketServer } from 'ws';
import * as Y from 'yjs';
import { setupWSConnection } from 'y-websocket/bin/utils';
import { Server, WebSocket } from 'ws';

@WebSocketGateway({ cors: true })
export class AppGateway implements OnGatewayConnection, OnGatewayDisconnect {
  @WebSocketServer()
  server: Server;

  handleConnection(client: WebSocket, ...args: any[]): void {
    console.log('connect');
    const request = args[0];
    const fileId = this.extractFileId(request.url);

    // Create a new Yjs document or get an existing one
    const ydoc = new Y.Doc();

    // Setup WebSocket connection for Yjs
    setupWSConnection(client, request, ydoc, {
      gc: true, // garbage collection
    });

    console.log(`Client connected for document ID: ${fileId}`);
  }

  handleDisconnect(client: WebSocket): void {
    console.log('Client disconnected');
  }

  private extractFileId(url: string): string {
    // Extract fileId from URL
    return url.split('/').pop();
  }
}
but not connected. is there any issue while using this with different servers. 

one more thing when using 
      provider.current = new WebsocketProvider(
        `wss://demos.yjs.dev/ws/`, // Replace with your WebSocket server URL
        `${fileId}`,
        ydoc.current,
      )
      
      its working.
